addons/stock_picking_batch/tests/test_batch_picking.py

```diff
@@ -327,3 +327,43 @@ def test_batch_with_immediate_transfer_and_backorder_wizard_with_manual_operatio
         # ensure that quantity for picking has been moved
         self.assertFalse(sum(quant_A.mapped('quantity')))
         self.assertFalse(sum(quant_B.mapped('quantity')))
+
+    def test_put_in_pack(self):
+        self.env['stock.quant']._update_available_quantity(self.productA, self.stock_location, 10.0)
+        self.env['stock.quant']._update_available_quantity(self.productB, self.stock_location, 10.0)
+
+        # Confirm batch, pickings should not be automatically assigned.
+        self.batch.action_confirm()
+        self.assertEqual(self.picking_client_1.state, 'confirmed', 'Picking 1 should be confirmed')
+        self.assertEqual(self.picking_client_2.state, 'confirmed', 'Picking 2 should be confirmed')
+        # Ask to assign, so pickings should be assigned now.
+        self.batch.action_assign()
+        self.assertEqual(self.picking_client_1.state, 'assigned', 'Picking 1 should be ready')
+        self.assertEqual(self.picking_client_2.state, 'assigned', 'Picking 2 should be ready')
+
+        # only do part of parkings + assign different destinations + try to pack (should get wizard to correct destination)
+        self.batch.move_line_ids.qty_done = 5
+        self.batch.move_line_ids[0].location_dest_id = self.stock_location.id
+        wizard_values = self.batch.put_in_pack()
+        wizard = self.env[(wizard_values.get('res_model'))].browse(wizard_values.get('res_id'))
+        wizard.location_dest_id = self.customer_location.id
+        package = wizard.action_done()
+
+        # a new package is made and done quantities should be in same package
```

Comment:
If I understood well how "cluster picking" works, you must have the same package id on all your move lines, not only on the ones with done qty. So I think it's better to check the `package_id` on all the move lines in this case.

Reply:
Ignore this comment, I was looking to different commit in the same time, you did it in the right commit, sorry.